### PR TITLE
Clarify setContext signature

### PR DESCRIPTION
**Pull request by lforst**
Fixes https://github.com/getsentry/sentry-docs/issues/5904

### CODE REVIEW

**Review by AbhiPrasad — APPROVED** [review]

File: src/platforms/common/enriching-events/context.mdx [hunk 1]
```diff
@@ -21,7 +21,7 @@ If you need to be able to search on custom data, [use tags](../tags) instead.
 
 ## Structured Context
 
-The best practice to attach custom data is via structured contexts. A context must always be a dictionary or map, and its values can be arbitrary.
+The best practice to attach custom data is via structured contexts. A context must always be a dictionary, object or map (depending on the programming language you're using), and its values can be arbitrary.
```
Comment (mydea) [inline src/platforms/common/enriching-events/context.mdx:24]: Should we just make this platform specific? E.g.:

```mdx
<PlatformSection supported={["javascript", "node"]}>
The best practice to attach custom data is via structured contexts. A context must always be an object, and its values can be arbitrary.
</PlatformSection>

<PlatformSection notSupported={["javascript", "node"]}>
The best practice to attach custom data is via structured contexts. A context must always be a dictionary or map, and its values can be arbitrary.
</PlatformSection>
```

No strong feelings, also OK this way but figured we may as well provide "correct" wording, and in this case the duplication is not too large. But feel free to discard!

File: src/platforms/common/enriching-events/context.mdx [hunk 2]
```diff
@@ -21,7 +21,15 @@ If you need to be able to search on custom data, [use tags](../tags) instead.
 
 ## Structured Context
 
-The best practice to attach custom data is via structured contexts. A context must always be a dictionary or map, and its values can be arbitrary.
+<PlatformSection supported={["javascript", "node"]}>
+  The best practice to attach custom data is via structured contexts. A context
+  must always be an object, and its values can be arbitrary.
```
Comment (shanamatthews) [inline src/platforms/common/enriching-events/context.mdx:26]: sorry, I can't help myself, english here is weird. This is better imo:

```suggestion
  The best way to attach custom data is with a structured context. A context
  must always be an object and its values can be arbitrary.
```

File: src/platforms/common/enriching-events/context.mdx [hunk 3]
```diff
@@ -21,7 +21,15 @@ If you need to be able to search on custom data, [use tags](../tags) instead.
...
+</PlatformSection>
+
+<PlatformSection notSupported={["javascript", "node"]}>
+  The best practice to attach custom data is via structured contexts. A context
+  must always be a dictionary or map, and its values can be arbitrary.
```
Comment (shanamatthews) [inline src/platforms/common/enriching-events/context.mdx:31]:
```suggestion
  The best way to attach custom data is with a structured context. A context
  must always be an object and its values can be arbitrary.
```

**Review by shanamatthews — APPROVED** [review]

**Comment by lforst** [comment]
Thanks @shanamatthews! :)

**Pull request merged**